org.adempiere.base/src/org/compiere/model/MToolBarButtonRestrict.java

```diff
@@ -61,6 +66,18 @@
 			+ " AND AD_Tab_ID=?"
 			+ " AND AD_ToolBarButton_ID IN" 
 			+ " (SELECT AD_ToolBarButton_ID FROM AD_ToolBarButton WHERE AD_Process_ID IS NULL AND IsActive='Y')";
+
+	// Toolbar button access restriction
+	public static final String SQL_IS_TOOLBAR_BTN_EXCLUDED = "SELECT tbr.IsExclude FROM AD_ToolBarButton tb 														"
+			+ " INNER JOIN AD_ToolBarButtonRestrict tbr ON 																											"
+			+ "						  	( tbr.AD_ToolBarButton_ID=tb.AD_ToolBarButton_ID AND tbr.AD_Client_ID IN (0,?) AND tbr.IsActive='Y' AND 					"
+			+ " 						  tbr.Action='W' AND (tbr.AD_Role_ID IS NULL OR tbr.AD_Role_ID=?) AND (tbr.AD_Window_ID IS NULL OR tbr.AD_Window_ID=?))		"
+			+ " LEFT JOIN AD_Tab t ON 	( t.AD_Tab_ID=tbr.AD_Tab_ID AND t.IsActive='Y' AND t.AD_Window_ID=? AND ((t.TabLevel=0 AND 0=?) OR t.AD_Tab_ID=?))			"
+			+ " WHERE tb.IsActive='Y' AND tb.Action='W' AND tb.Name=? AND tb.AD_Tab_ID IS NULL AND (tbr.AD_Tab_ID IS NULL OR t.AD_Tab_ID IS NOT NULL)				"
+			+ " ORDER BY tbr.AD_Client_ID DESC, tbr.AD_Window_ID, tbr.AD_Tab_ID 																					";
+
+	public static final String	TOOLBAR_BTN_WINDOW_NEW	= "Window - New";
```

Comment:
This creates a weak dependency on AD_ToolbarButton.Name="Window - New".
If somebody changes that name the functionality will not work anymore.

I would advise better to create the dependency with the AD_ToolbarButton_ID=200031 (added in SystemIDs).